### PR TITLE
Fix/close drawer

### PR DESCRIPTION
**Pull request by lbebber**
#### What is the purpose of this pull request?
Closes drawer when a link is clicked.

Fixes issue described [here](https://github.com/vtex-apps/render-runtime/pull/470). That fix is going to be reverted temporarily [by this PR](https://github.com/vtex-apps/render-runtime/pull/490) due to unforeseen bugs in third party components.

Test on https://lbebber--joythestore.myvtex.com/, on mobile mode. Go to the drawer, click on an item on "womenswear", and then open the menu again and click on another element.

#### Types of changes

* [ ] Bug fix (a non-breaking change which fixes an issue)
* [ ] New feature (a non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/Drawer.tsx [hunk 1]
```diff
@@ -69,6 +75,42 @@ const CSS_HANDLES = [
   'closeIconButton',
 ]
 
+// This is a totally valid use case for any, eslint.
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function isHTMLElement(x: any): x is HTMLElement {
+  return !!x.tagName
+}
+
+const isLink = (element: HTMLElement) => {
```
Comment (Klynger) [inline react/Drawer.tsx]:
```suggestion
const isLink = (element: HTMLElement): element is HTMLAnchorElement => {
```

File: react/Drawer.tsx [hunk 2]
```diff
@@ -69,6 +75,42 @@ const CSS_HANDLES = [
   'closeIconButton',
 ]
 
+// This is a totally valid use case for any, eslint.
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function isHTMLElement(x: any): x is HTMLElement {
+  return !!x.tagName
+}
```
Comment (Klynger) [inline react/Drawer.tsx]:
```suggestion
function isHTMLElement(x: any): x is HTMLElement {
  return 'tagName' in x
}
```

File: react/Drawer.tsx [hunk 3]
```diff
@@ -79,9 +121,32 @@ const Drawer: StorefrontComponent<
...
+    const element = childrenContainer.current
+
+    if (!element) {
+      return
+    }
+
+    const handleContainerClick = (event: MouseEvent) => {
+      const { target } = event
+
+      if (isElementInsideLink(target as HTMLElement, element)) {
```
Comment (Klynger) [inline react/Drawer.tsx]:
```suggestion
    const handleContainerClick = (event: MouseEvent<HTMLElement>) => {
      const { target } = event

      if (isElementInsideLink(target, element)) {
```

Comment (lbebber) [reply react/Drawer.tsx]: boa a intençao, mas nao funciona (MouseEvent nao é generico)

File: react/Drawer.tsx (hunk 2 above)
Comment (lbebber) [reply react/Drawer.tsx]: boa

File: react/Drawer.tsx [hunk 4]
```diff
@@ -133,7 +198,10 @@ const Drawer: StorefrontComponent<
                   <IconClose size={30} type="line" />
                 </button>
               </div>
-              <div className={`${handles.childrenContainer} flex flex-grow-1`}>
+              <div
+                className={`${handles.childrenContainer} flex flex-grow-1 b--red`}
```
Comment (victorhmp) [inline react/Drawer.tsx]: Why did you add `b--red`?

File: react/Drawer.tsx (hunk 3 above)
Comment (lbebber) [reply react/Drawer.tsx]: `MouseEvent` nativo nao é generico, mas `React.MouseEvent` é, assim como `ReactEventHandler` (@Klynger e eu conversamos, só deixando registrado)

File: react/Drawer.tsx [hunk 5]
```diff
@@ -69,6 +75,42 @@ const CSS_HANDLES = [
...
 ]
 
+// This is a totally valid use case for any, eslint.
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function isHTMLElement(x: any): x is HTMLElement {
+  return !!x.tagName
+}
+
+const isLink = (element: HTMLElement) => {
+  return String(element.tagName).toUpperCase() === 'A'
```
[the remark this answers is not in the document]
Comment (lbebber) [reply react/Drawer.tsx]: just as a guarantee 🤷‍♂

File: react/Drawer.tsx (hunk 4 above)
Comment (lbebber) [reply react/Drawer.tsx]: ooops good catch--it was for debugging purposes
